.github/workflows/python_unit_test.yaml

```diff
@@ -28,22 +39,28 @@ jobs:
           --health-interval 10s
           --health-timeout 5s
           --health-retries 5
-    
+
     steps:
       - name: Check out code
         uses: actions/checkout@v3
       - name: Set up python
         uses: actions/setup-python@v4
         with:
-          python-version: "3.6.7"
+          python-version: ${{ inputs.python-version }}
       - name: Install poetry
         run: |
-            pip --version
-            pip install poetry
-            poetry install -vv
+          pip --version
+          pip install poetry
+          poetry install -vv
       - name: Service specific setup
         run: |
-            curl -sSL "${{inputs.setup-script-location}}" | bash
+          curl -sSL "${{inputs.setup-script-location}}" | bash
       - name: Run test
-        run: | 
-          poetry run pytest -vv --cov=fence --cov=migrations/versions --cov-report xml tests 
+        run: |
+          poetry run pytest -vv --cov=${{ env.REPO_NAME }} --cov=migrations/versions --cov-report xml tests
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v2
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
i haven't seen codecov used before in our repos, we usually use coveralls (and/or codacy for older repos). And where is `CODECOV_TOKEN` defined?

I think this step needs to be more flexible too, to support other coverage solutions as well as _no_ coverage solution